### PR TITLE
Support account locking feature for backup codes

### PR DESCRIPTION
**Pull request by AnuradhaSK**
## Purpose
Support account locking feature for backup codes

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java [hunk 1]
```diff
@@ -320,27 +320,29 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
...
                 if (log.isDebugEnabled()) {
                     log.debug("Given otp code is mismatch.");
                 }
-                handleOtpVerificationFail(context);
-                throw new AuthenticationFailedException("Code mismatch.");
             }
         }
+        if (!succeededAttempt) {
+            handleOtpVerificationFail(context);
+            throw new AuthenticationFailedException("Code mismatch.");
```
Comment (somindatommy) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: Is it possible to throw an error code here?

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java [hunk 2]
```diff
@@ -359,16 +361,18 @@ private void processValidUserToken(AuthenticationContext context, AuthenticatedU
...
-     * Checks the backup codes for email otp.
+     * Check whether the entered code matches with a backup code.
      *
-     * @param context           AuthenticationContext.
-     * @param userToken         EmailOTP token.
-     * @param authenticatedUser AuthenticatedUser
-     * @throws AuthenticationFailedException
+     * @param context The AuthenticationContext.
+     * @param userToken The userToken.
+     * @param authenticatedUser The authenticatedUser.
```
Comment (somindatommy) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: Shall we fix formatting in the method comment?

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java (hunk 1 above)
Comment (AnuradhaSK) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: There is no constructor in `AuthenticationFailedException` to pass an error code

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java (hunk 2 above)
Comment (AnuradhaSK) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: Fixed in https://github.com/wso2-extensions/identity-outbound-auth-email-otp/pull/84/commits/04b1581704622d8e7af97178e36c0406370a951d

**Review by somindatommy — APPROVED** [review]

**Pull request merged**